### ISSUE TITLE
While training the model, GPU util reaches 100% but no progress happends.

### ISSUE THREAD

**Issue by nickle-fang** [pytorch/pytorch#49669]
## ❓ While training my model, the GPU util reaches 100% but no progress happends. 
Just as the title says. And the code stucks at "image.to(device)". But when I comment the line "loss.backward()", the problem disappears. 
My environment is as follows:
- cuda 11.0
- pytorch 1.7.1 
- python 3.6.9

**Comment by nickle-fang** [comment]
And strangely, the model can process 4 pictures before stuck.

**Comment by grenkoca** [comment]
What's your GPU memory usage at? Also it would be helpful if you could post your code.

**Comment by ngimel** [comment]
Please provide self-contained script to reproduce your problem.

**Comment by nickle-fang** [comment]
> What's your GPU memory usage at? Also it would be helpful if you could post your code.

The GPU memory usage is at about 4000MB while total memory is 8000MB.

**Comment by nickle-fang** [comment]
> Please provide self-contained script to reproduce your problem.

Thanks and I will upload the script later!
